### PR TITLE
Adding C bubble sort

### PR DESCRIPTION
**Pull request by Gathros**

### CODE REVIEW

File: chapters/sorting_searching/bubble/code/c/bubble_sort.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#include <stdio.h>
+
+void bubble_sort(int *array, size_t n) {
```
Comment (ubsan) [inline chapters/sorting_searching/bubble/code/c/bubble_sort.c]: I think you should probably include `stddef.h` here.

File: chapters/sorting_searching/bubble/code/c/bubble_sort.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        if (!swapped) {
+            break;
+        }
+    }
+}
+
+int main() {
+    int array[10] = {1, 45, 756, 4569, 56, 3, 8, 5, -10, -4};
```
Comment (ubsan) [inline chapters/sorting_searching/bubble/code/c/bubble_sort.c]: Could you use a `#define`  instead of magic constants throughout the code?

Comment (Gathros) [reply chapters/sorting_searching/bubble/code/c/bubble_sort.c]: I don't think that's needed really, it's just an example.

**Comment by leios** [comment]
Same as in #80 , I think `array[10]` is fine for this case. If there are no other issues, I'll go ahead and merge it.

**Comment by leios** [comment]
@ubsan I do agree that a `#define n 10` would help a bit for readability in both cases, actually.

**Comment by Gathros** [comment]
@leios I get that but it's a part of the example and not the implementation of bubble sort.

**Comment by leios** [comment]
yeah, I get that too. I think it's probably fine this way.

**Comment by leios** [comment]
With this and #80, I think the `array[10]` is fine for now specifically because it's example code.

**Pull request merged**